aggregator/internal/pkg/server.go

```diff
@@ -35,6 +35,48 @@ func (agg *Aggregator) ServeOperators() error {
 	return err
 }
 
+// Waits for the arrival of task associated with signedTaskResponse and returns true on success or false on failure
+// If the task is not present in the internal map, it will try to fetch it from logs and retry.
+// The number of retries is specified by `waitForEventRetries`, and the waiting time between each by `waitForEventSleepSeconds`
+func (agg *Aggregator) waitForTaskAndFetchIfLost(signedTaskResponse *types.SignedTaskResponse) bool {
```

Comment:
This should be splitted, I also changed the name of the PR to reflect more the real issue, since we have another one that does something similar. Keep the old ```waitForTaskResponses```, and simply make a function that fetches a task response if you don't have it when an operator answers.

Reply:
It also returns a boolean but has no indication of what it means